files/image_config/kubernetes/kubernetes_master_entrance.sh

```diff
@@ -1,2 +1,23 @@
+#!/bin/bash
+
 # This script is for kubernetes master image usage
 # Will mount kubernetes master disk and execute kubernetes entrance script
+
+set -x
+
+# mount disk from host
+mount_point="/from_host"
+disk="/dev/sdb1"
+mkdir -p $mount_point
+mount $disk $mount_point
+# check whether it is the first time to boot
+first_boot_flag_file="/from_host/first_boot_flag"
```

Comment:
>first_boot_flag [](http://example.com/codeflow?start=33&length=15)

Is this a new feature?
Wondering if you can use existing sonic feature:
```
FIRST_BOOT_FILE="/host/image-${SONIC_VERSION}/platform/firsttime"
```

Reply:
Just want to check whether it's the first boot here, if it's the first time, need to call the master script. If there is already one same feature, I can reuse the existing one. But after I checked, I don't find there is this flag ( "/host/image-${SONIC_VERSION}/platform/firsttime") both at first boot and subsequent restart

Reply:
I mean it's not good to use the existing feature. The reason is that we call the k8s master script in the last line of rc.local, but we don't change the rc.local file directly. We insert the call master script line by sed command when include_kubernetes_master=y. If we want to add the calling k8s master script to existing firsttime flag logic, we need to change the rc.local file directly, it's not good to add lines to rc.local by sed command any more. If we agree to change the rc.local file directly, I can try to use the existing flag.

Reply:
I see the point of `first_boot_flag`, `kubernetes_master_entrance.sh` and `sed -i '/^exit 0/i\bash /usr/sbin/kubernetes_master_entrance.sh' $FILESYSTEM_ROOT/etc/rc.local`.

They are introduced for a deployment workflow, but this workflow is fragile. Suggest:
1. Create and start the Kubernetes master VM
2. Configure the VM IP by hyperv-daemons, or by console access
3. SSH into the VM, so you have full control
4. Copy some scripts into the VM
5. Enable/start some services.

Reply:
I created https://github.com/sonic-net/sonic-buildimage/issues/15849, and please consider it in future PR.